### PR TITLE
Support for  matryoshka embeddings

### PR DESCRIPTION
**Pull request by wirthual**
## Related Issue

#476 

## Checklist
- [x] I have read the [CONTRIBUTING](https://github.com/michaelfeil/infinity/tree/main?tab=readme-ov-file#contribute-and-develop) guidelines.
- [x] I have added tests to cover my changes.
- [x] I have updated the documentation (docs folder) accordingly.

## Additional Notes
WIP to add matryoshka embeddings.

Is there a CLAP model which supports matryoshka embedding for testing?
Is there a TinyCLIP model which supoprts matryoshka embedding for testing?

Currently missing:
  [  ] Integration into client
  [  ] Implementation for dummy model

### CODE REVIEW

File: libs/infinity_emb/infinity_emb/fastapi_schemas/pymodels.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ class _OpenAIEmbeddingInput(BaseModel):
     model: str = "default/not-specified"
     encoding_format: EmbeddingEncodingFormat = EmbeddingEncodingFormat.float
     user: Optional[str] = None
+    dimensions: Optional[int] = None
```
Comment (michaelfeil) [inline libs/infinity_emb/infinity_emb/fastapi_schemas/pymodels.py:57]: int should be `0 < x < 8193`

**Comment by michaelfeil** [comment]
LGTM, if you change the OpenAPI spec for the validation of input.

**Comment by michaelfeil** [comment]
@wirthual 
Potentially this. could be a end-to-end test!

```
from openai import OpenAI

client = OpenAI(
    base_url="http://0.0.0.0:7997",
    api_key="sk", 
)

result = client.embeddings.create(
    input=["input","input2"],
    model="nomic-ai/nomic-embed-text-v1.5",
    dimensions=64
)

assert len(result.data[0].embedding) == 64
```

**Comment by wirthual** [comment]
> @wirthual Potentially this. could be a end-to-end test under open
> 
> ```
> from openai import OpenAI
> 
> client = OpenAI(
>     base_url="http://0.0.0.0:7997",
>     api_key="sk", 
> )
> 
> result = client.embeddings.create(
>     input=["input","input2"],
>     model="nomic-ai/nomic-embed-text-v1.5",
>     dimensions=64
> )
> 
> assert len(result.data[0].embedding) == 64
> ```

Sounds good. Is there an exmaple on how to start a fastapi server within a pytest method without using `AsyncOpenAI` client?

**Comment by michaelfeil** [comment]
Just add one here: 
https://github.com/michaelfeil/infinity/blob/be483785f23c3e2a738c85028cbac3a390ec2bab/libs/infinity_emb/tests/end_to_end/test_openapi_client_compat.py#L115C9-L115C21
Also with the other tests - mostly we don't use `pytest.mark.parametrize` here to that it does not need to restart the server every time.

**Comment by wirthual** [comment]
Like [this](https://github.com/michaelfeil/infinity/blob/9c811fbc7bebb4c09f26fe3e7ca19a27ced52c1a/libs/infinity_emb/tests/end_to_end/test_openapi_client_compat.py#L124)?

**Review by michaelfeil — APPROVED** [review]
Okay, nevermind..  :)

**Pull request merged**